tools/python/runfiles/BUILD

```diff
@@ -20,7 +21,14 @@ filegroup(
 py_library(
     name = "runfiles",
     testonly = 1,
-    srcs = ["runfiles.py"],
+    srcs = [
+        "runfiles.py",
+        ":runfiles_constants",
+    ],
+)
+
+gen_runfiles_constants(
+    name = "runfiles_constants",
```

Comment:
The MAIN_REPOSITORY_RUNFILES_DIRECTORY looks very tempting to use (e.g. "if CurrentRespository() == MAIN_RUNFILES_DIRECTORY"); let's try to make it more explicit that's private.

Please have the file name generated have a leading underscore ("_runfiles_constants"), or otherwise have a more internal-sounding name.

Reply:
Leading underscore is sufficient.